src/dscanner/analysis/objectconst.d

```diff
@@ -5,100 +5,89 @@
 	mixin AnalyzerInfo!"object_const_check";
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
-	///
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName);
 	}
 
-	mixin visitTemplate!ClassDeclaration;
-	mixin visitTemplate!InterfaceDeclaration;
-	mixin visitTemplate!UnionDeclaration;
-	mixin visitTemplate!StructDeclaration;
-
-	override void visit(const AttributeDeclaration d)
+	void visitAggregate(AST.AggregateDeclaration ad)
 	{
-		if (d.attribute.attribute == tok!"const" && inAggregate)
-		{
-			constColon = true;
-		}
-		d.accept(this);
-	}
+		import dmd.astenums : MODFlags, STC;
 
-	override void visit(const Declaration d)
-	{
-		import std.algorithm : any;
-		bool setConstBlock;
-		if (inAggregate && d.attributes && d.attributes.any!(a => a.attribute == tok!"const"))
-		{
-			setConstBlock = true;
-			constBlock = true;
-		}
-
-		bool containsDisable(A)(const A[] attribs)
-		{
-			import std.algorithm.searching : canFind;
-			return attribs.canFind!(a => a.atAttribute !is null &&
-				a.atAttribute.identifier.text == "disable");
-		}
+		if (!ad.members)
+			return;
 
-		if (const FunctionDeclaration fd = d.functionDeclaration)
+		foreach(member; *ad.members)
 		{
-			const isDeclationDisabled = containsDisable(d.attributes) ||
-				containsDisable(fd.memberFunctionAttributes);
-
-			if (inAggregate && !constColon && !constBlock && !isDeclationDisabled
-					&& isInteresting(fd.name.text) && !hasConst(fd.memberFunctionAttributes))
+			if (auto fd = member.isFuncDeclaration())
+			{
+				if (fd.type && !fd.type.mod == MODFlags.const_ && isInteresting(fd.ident.toString()) && 
+					!(fd.storage_class & STC.disable))
+						addErrorMessage(cast(ulong) fd.loc.linnum, cast(ulong) fd.loc.charnum, KEY,
+							"Methods 'opCmp', 'toHash', 'opEquals', 'opCast', and/or 'toString' are non-const.");
+				
+				member.accept(this);
+			}
+			else if (auto scd = member.isStorageClassDeclaration())
 			{
-				addErrorMessage(d.functionDeclaration.name.line,
-						d.functionDeclaration.name.column, "dscanner.suspicious.object_const",
-						"Methods 'opCmp', 'toHash', 'opEquals', 'opCast', and/or 'toString' are non-const.");
+				foreach (smember; *scd.decl)
+				{
+					if (auto fd2 = smember.isFuncDeclaration())
+					{
+						if (fd2.type && !fd2.type.mod == MODFlags.const_ && isInteresting(fd2.ident.toString()) && 
+							!(fd2.storage_class & STC.disable) && !(scd.stc & STC.const_ ||
+							scd.stc & STC.immutable_ || scd.stc & STC.wild))
+								addErrorMessage(cast(ulong) fd2.loc.linnum, cast(ulong) fd2.loc.charnum, KEY,
+									"Methods 'opCmp', 'toHash', 'opEquals', 'opCast', and/or 'toString' are non-const.");
+						
+						smember.accept(this);
+					}
+					else
+						smember.accept(this);
+				}
 			}
+			else
+				member.accept(this);
 		}
+	}
 
-		d.accept(this);
+	override void visit(AST.ClassDeclaration cd)
+	{
+		visitBaseClasses(cd);
```

Comment:
I don't think this method is needed. Baseclasses are also classes, therefore the transitive visitor will check them at some point.

Reply:
This method is called in trasitive visitor when visiting AST.ClassDeclaration. I think it is needed there because if we override the visit(AST.ClassDeclaration ) method and do not include visitBaseClasses, the second method will end up never being called

Reply:
Have you tried it?